src/main/java/io/debezium/connector/db2/Db2ValueConverters.java

```diff
@@ -52,6 +52,10 @@ public SchemaBuilder schemaBuilder(Column column) {
             case Types.TINYINT:
                 // values are an 8-bit unsigned integer value between 0 and 255, we thus need to store it in short int
                 return SchemaBuilder.int16();
+            case Types.OTHER:
+                if (matches(column.typeName().toUpperCase(), "DECFLOAT")) {
+                    return SchemaBuilder.float64();
```

Comment:
What is the real datatype coming from the database? It seems to me according to docs that it is either 16 or 32 bit signed floating point decimal. How does it differ from REAL/FLOAT? Does it makes sense to map it to either FLOAT32 or FLOAT64 based on the value? What Java class is returned by the JDBC driver when small and large DECFLOATs are used?

Reply:
The range of a decimal floating point number is either 16 or 34 digits of precision. You can refer to this manual. https://www.ibm.com/docs/en/db2-for-zos/12?topic=numbers-decimal-floating-point-decfloat

Create table like this,
create table TEST.T1
(
    ID INTEGER not null primary key,
    C_DECFLOAT DECFLOAT(34)
);

Then insert a record, INSERT INTO TEST.T1 (ID, C_DECFLOAT) VALUES (1, DECFLOAT(1.234));
Finally execute the select statement，select C_DECFLOAT from TEST.T1.  The result is 1.234, without padding the 0's at the end.

To be honest, I don’t quite understand the difference between decfloat and float types.